### PR TITLE
Adding Validation In Interceptor

### PR DESCRIPTION
**Pull request by malinthaprasan**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/interceptors/ValidationInInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            ExecutableValidator methodValidator = validator.forExecutables();
+            Set<ConstraintViolation<T>> violations = methodValidator.validateParameters(instance,
+                    method, arguments);
+
+            if (!violations.isEmpty()) {
+                throw new ConstraintViolationException(violations);
+            }
+
+        } catch (ValidationException ex) {
+            throw new InternalServerErrorException(ex);
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/interceptors/ValidationInInterceptor.java]: Please use proper exception handling mechanisms.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/exception/ConstraintViolationException.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+public class ConstraintViolationException extends WebApplicationException {
+
+    protected Log log = LogFactory.getLog(getClass());
+
+    public <T> ConstraintViolationException(Set<ConstraintViolation<T>> violations) {
+        super(Response.status(Response.Status.BAD_REQUEST).entity(
+                RestApiUtil.getConstraintViolationErrorDTO(violations)).build());
+        for (ConstraintViolation violation : violations) {
+            log.info(violation.getRootBeanClass().getSimpleName() + "." + violation.getPropertyPath()
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/exception/ConstraintViolationException.java]: Info logs removed in #1474 (https://github.com/wso2/carbon-apimgt/commit/88fc541f8c1d85f3fb0edc657109913cc2c04e08)

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/interceptors/ValidationInInterceptor.java (hunk 1 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/interceptors/ValidationInInterceptor.java]: Fixed in 1528 (https://github.com/wso2/carbon-apimgt/commit/ab4f846a67ce38c739e8c44dbb462ba844e65bb6)

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/interceptors/ValidationInInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            }
+
+            Set<ConstraintViolation<T>> violations = validator.validate(object);
+
+            if (!violations.isEmpty()) {
+                throw new ConstraintViolationException(violations);
+            }
+
+        } catch (ValidationException ex) {
+            throw new InternalServerErrorException(ex);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/interceptors/ValidationInInterceptor.java]: Removed try-catch block in #1528 (https://github.com/wso2/carbon-apimgt/commit/ab4f846a67ce38c739e8c44dbb462ba844e65bb6) as it was not mandatory.

**Pull request merged**